client/src/services/notificationManager.ts

```diff
@@ -40,105 +22,26 @@ export interface NotificationOptions {
   trace?: boolean;
 }
 
-export interface NotificationToastOptions extends NotificationOptions {
-  duration?: number;
-}
-
-const DEFAULT_NOTIFICATION_DURATION = 5000;
-
 export class NotificationManager {
   notifications: Ref<Notification[]>;
   unreadCount: Ref<number> = computed(() => this.notifications.value.filter((n) => n.read === false).length);
-  toastManager: ToastManager;
 
   constructor() {
     this.notifications = ref([]);
-    this.toastManager = new ToastManager();
-  }
-
-  async showModal(notification: Notification, options?: NotificationOptions): Promise<void> {
-    if (options && options.addToList) {
-      this.addToList(notification);
-    }
-    if (options && options.trace) {
-      this._trace(notification);
-    }
-
-    const alertModalConfig: MsAlertModalConfig = {
-      theme: this._getMsReportTheme(notification.level),
-      title: notification.title,
-      message: notification.message,
-    };
-
-    const result = await this._createAndPresentModal(alertModalConfig);
-    if (result && result.role === MsModalResult.Confirm) {
-      this.markAsRead(notification.id);
-    }
-  }
-
-  async _createAndPresentModal(modalConfig: MsAlertModalConfig): Promise<any> {
-    const top = await modalController.getTop();
-    if (top) {
-      top.classList.add('overlapped-modal');
-    }
-
-    const modal = await modalController.create({
-      component: MsAlertModal,
-      canDismiss: true,
-      cssClass: 'notification-modal',
-      componentProps: modalConfig,
-    });
-    await modal.present();
-    const result = await modal.onWillDismiss();
-    await modal.dismiss();
-
-    if (top) {
-      top.classList.remove('overlapped-modal');
-    }
-    return result;
   }
 
-  async showToast(notification: Notification, options?: NotificationToastOptions): Promise<void> {
-    if (options && options.addToList) {
-      this.addToList(notification);
-    }
-    if (options && options.trace) {
-      this._trace(notification);
-    }
-
-    this.toastManager
-      .createAndPresent({
-        theme: this._getMsReportTheme(notification.level),
-        title: notification.title,
-        message: notification.message,
-        duration: DEFAULT_NOTIFICATION_DURATION,
-      })
-      .then(async (result) => {
-        if (result && result.role === 'confirm') {
-          this.markAsRead(notification.id);
-        }
-      });
-  }
-
-  private _getMsReportTheme(notificationLevel: NotificationLevel): MsReportTheme {
-    switch (notificationLevel) {
-      case NotificationLevel.Info:
-        return MsReportTheme.Info;
-      case NotificationLevel.Success:
-        return MsReportTheme.Success;
-      case NotificationLevel.Warning:
-        return MsReportTheme.Warning;
-      case NotificationLevel.Error:
-        return MsReportTheme.Error;
-    }
+  create(information: Information): Notification {
```

Comment:
Is there a point in returning the object? Why not just call this method `addNotification()` or something like that and avoid exposing an object that is only used internally?

Reply:
At the moment, I'm not sure we need to return it yh.
However, the `create` name is the more logical looking at what is doing the function. It takes an Information and create a Notification.
Instead, `addNotification` is quite disturbing on multiple points:
 - it creates it, not only 'add' it
 - repeating notification on a notificationManager seems useless: `notificationManager.addNotification()` we already know it's a notification thanks to the manager name.
 Maybe only `add` could be a compromise even if I don't see the point, since `create` is perfect match here?

Reply:
`create` to me takes a more factory connotation: I'm creating an object and returning it.

Here we indeed create it, but we're also adding it to a list, and returning it seems pointless. But I'm not going to die on this hill, it doesn't matter much.

Reply:
That's okay to me, `add` seems what we want if we as you said don't return it yh.